src/api/postAmmFundingOrder.js

```diff
@@ -0,0 +1,51 @@
+const { Joi } = require('dvf-utils')
+const R = require('ramda')
+
+const post = require('../lib/dvf/post-authenticated')
+const validateWithJoi = require('../lib/validators/validateWithJoi')
+
+const nonNegativeInteger = Joi.number().integer().min(0)
+const maxTimestampHours = Math.pow(2, 22) * 1000 * 60 * 60 - 1
+
+// NOTE: this schema should be shared between backend and client.
+const starkOrderSchema = Joi.object({
+  vaultIdSell: nonNegativeInteger,
+  vaultIdBuy: nonNegativeInteger,
+  amountSell: Joi.string(),
+  amountBuy: Joi.string(),
+  tokenSell: Joi.string(),
+  tokenBuy: Joi.string(),
+  // If not provided, random nonce will be generated for each order.
+  nonce: nonNegativeInteger.optional(),
+  // If not provided, expirationTimestamp will be set based on
+  // dvf.config.defaultStarkExpiry.
+  expirationTimestamp: nonNegativeInteger.max(maxTimestampHours).optional(),
+  signature: Joi.object().optional().keys({
+    v: Joi.string().optional(),
+    r: Joi.string(),
+    s: Joi.string()
+  })
+// If orders are pre-signed, nonce and expirationTimestamp needs to be present.
+}).with('signature', ['nonce', 'expirationTimestamp'])
+
+const schema = Joi.object({
+  pool: Joi.string(),
+  orders: Joi.array()
+    .items(Joi.object({
+      starkOrder: starkOrderSchema
+    }))
+    .length(2)
+})
+
+const validateData = validateWithJoi(schema)('INVALID_METHOD_ARGUMENT')({
+  context: 'postAmmFundingOrder'
+})
+
+const endpoint = '/v1/trading/amm/postAmmFundingOrder'
+
+module.exports = async (dvf, data, nonce, signature) => post(
+  dvf, endpoint, nonce, signature, R.compose(
+    dvf.stark.singAmmFundingOrder,
```

Comment:
is this meant to be `sign` instead of `sing`?

Reply:
fixed in bba3f9f49865cfe56a72b8f516231b5433754c08